clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -122,6 +123,48 @@ public void setup() {
         this.props.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
     }
 
+    @Test
+    public void testOffsetFetchRequestStateToStringBase() {
+        ConsumerConfig config = mock(ConsumerConfig.class);
+        CommitRequestManager.MemberInfo memberInfo = new CommitRequestManager.MemberInfo();
+
+        CommitRequestManager commitRequestManager = new CommitRequestManager(
+                time,
+                logContext,
+                subscriptionState,
+                config,
+                coordinatorRequestManager,
+                offsetCommitCallbackInvoker,
+                "groupId",
+                Optional.of("groupInstanceId"),
+                metrics);
+
+        commitRequestManager.onMemberEpochUpdated(Optional.of(1), Optional.empty());
+        Set<TopicPartition> requestedPartitions = Collections.singleton(new TopicPartition("topic-1", 1));
+
+        CommitRequestManager.OffsetFetchRequestState offsetFetchRequestState = commitRequestManager.new OffsetFetchRequestState(
+                requestedPartitions,
+                retryBackoffMs,
+                retryBackoffMaxMs,
+                1000,
+                memberInfo);
+
+        TimedRequestState timedRequestState = new TimedRequestState(
+                logContext,
+                "CommitRequestManager",
+                retryBackoffMs,
+                retryBackoffMaxMs,
+                TimedRequestState.deadlineTimer(time, 0)
+        );
+
+        String target = timedRequestState.toStringBase() +
+                ", " + memberInfo +
+                ", requestedPartitions=" + offsetFetchRequestState.requestedPartitions;
+
+        assertDoesNotThrow(timedRequestState::toString);
+        assertEquals(target, offsetFetchRequestState.toStringBase());
+    }
```

Comment:
Maybe we can add `assertFalse(target.contains("Optional"));` to make sure we unwrap the optional variables?

Reply:
Added!